src/robot/libraries/String.py

```diff
@@ -462,6 +462,24 @@ def split_string_to_characters(self, string):
         """
         return list(string)
 
+    def split_string_by_delimiter(self, string, delimiter):
+        """Splits the given ``string`` by delimiter and returns a list of strings.
+
+        Example:
+        | @{string_split}= | Split String By Delimiter | ${string} | -
+
+        """
+        return string.split(delimiter)
+
+    def remove_leading_spaces(self, string):
+        """Removes any and only leading spaces in a string. Does not remove any spaces in between strings such as "Hello Hello"
+
+        Example:
+        | ${removed_leading_spaces}= | Remove Leading Spaces | ${string}
+
+        """
+        return string.split(string)
```

Comment:
The docstring does not match what the implementation does.

This code will basically return a list with two empty items in it and not remove the leading whitespace characters.

```
>>> f = "   \t   foo   "
>>> f.split(f)
['', '']
```

As a result, I think this should be `return string.lstrip()`

```
>>> f = "   \t   foo   "
>>> f.lstrip()
'foo   '
>>> f = "d   \t   foo   "
>>> f.lstrip()
'd   \t   foo   '
```